PeerioTechnologies/peerio-desktop
# feat: new upload dialog

#### Relevant info and issue/PR links 
main: https://app.clubhouse.io/peerio/story/220/file-preview-and-review-on-upload-in-chat
desktop-specific: https://app.clubhouse.io/peerio/story/357/desktop-file-preview-and-review-on-upload-in-chat

copy update in icebear: https://github.com/PeerioTechnologies/peerio-icebear/pull/111

#### Testing instructions  
Current status:
* new "file preview" dialog pops up when user uploads a file into a chat
  * image preview shows for images, file icon shows for non-images
  * allows user to rename file before upload
  * allows user to send message along with file
    * **TODO** currently waiting on updated SDK to ensure that message sends with file, not immediately
* works with both drag-and-drop and "Upload and share" button
* works with single and multiple uploads
* "edit" button opens secondary dialog to allow user to select other chats to share this file with
  * **TODO** doesn't work right now, only markup

Other changes:
* minor styling fixes e.g. RTB overrides, Avatar
* new file type icons, plus new icon sizes `large` and `xlarge`

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
